src/daemon.rs

```diff
@@ -43,6 +69,39 @@ pub async fn run(config: Config) -> Result<()> {
             _ = liveness_interval.tick() => {
                 println!("spiffe-helper-rust daemon is alive");
             }
+            // Watch for updates from X509Source
+            res = update_channel.changed() => {
+                 match res {
+                     Ok(_) => {
+                         println!("Received X.509 update notification");
+                         // Fetch and write
+                         match source.get_svid() {
+                             Ok(Some(svid)) => {
+                                 match source.get_bundle_for_trust_domain(svid.spiffe_id().trust_domain()) {
+                                     Ok(Some(bundle)) => {
+                                         if let Err(e) = workload_api::on_x509_update(&svid, &bundle, &config).await {
+                                             eprintln!("Failed to handle X.509 update: {}", e);
```

Comment:
Over indented code is bad code smell. This code looks very ugly and difficult to read @cursoragent